### ISSUE TITLE
Test Support for Apollo 3.0 beta

### ISSUE THREAD

**Issue by wtrocki** [apollographql/apollo-cache-persist#317]
As in topic new Apollo client exists as a single package with embedded cache

**Comment by wtrocki** [comment]
Apollo-Cache-Persist works quite nicely with the beta Apollo 3.0 after changing imports.
Leaving open to monitor the next releases. 

CC @benjamn

**Comment by dcecile** [comment]
3.0 doesn't have defaults anymore, only the `writeData` API. What's the best way to do "write data only once if not yet persisted"?

**Comment by wtrocki** [comment]
@dcecile Can you clarify this better. I would recommend to drop some separate issue if you have it running with 3.0

**Comment by dcecile** [comment]
@wtrocki No problem, I've tried to clarify this in #320

**Comment by tylerjbainbridge** [comment]
Hi @wtrocki- can you clarify which imports (related to this package) need to be updated? I'm in the process of updating to apollo 3.0 and having some trouble.

> Apollo-Cache-Persist works quite nicely with the beta Apollo 3.0 after changing imports.

**Comment by wtrocki** [comment]
This information is outdated. In latest beta there is common package that should be used along with the cache interface

**Comment by goofiw** [comment]
3.0 released today, has anyone tried spinning this up?

**Comment by Titozzz** [comment]
I'm trying to update, persist cache complains about the missing writeData API. Currently investigating whether this is fully breaking or not.

**Comment by Titozzz** [comment]
I've checked writeApi is not used in apollo-cache-persist and proceeded to @ts-expect-error for now. No issues YET.

**Comment by darrylyoung** [comment]
Thanks, @Titozzz. Glad to hear you're having no issues. Has anyone else tried this with Apollo Client 3.x yet? I'm planning on migrating to 3.0 this week and wondered what I'm in for regarding this package. 😄

**Comment by alessandro-bottamedi** [comment]
I'm using apollo-cache-persist since Apollo client Beta 35, no issues! 😄😄

**Comment by robertvorthman** [comment]
Here is an example of how I am using apollo-cache-persist with @apollo/client 3.0.2 in my web app.

```jsx
import React from 'react';
import { useEffect, useState } from 'react';
import { ApolloProvider } from '@apollo/client';
import { persistCache } from 'apollo-cache-persist';

//import a configured ApolloClient and InMemoryCache object
import client, { cache } from './client';

export default function ApolloProviderPersisted(props) {

  const [isPersisted, setIsPersisted] = useState();

  useEffect(() => {

    async function hydrate() {
      await persistCache({
        cache,
        storage: window.localStorage
      });
      setIsPersisted(true);
    }

    hydrate();

  }, []);

  if (!isPersisted) {
    //persistGate prop is a React component that displays a loading placeholder
    return props.persistGate || null;
  }

  return <ApolloProvider client={client}>{props.children}</ApolloProvider>

};
```

**Comment by punksta** [comment]
There are typescript problems due to changes in imports.

```
node_modules/apollo-cache-persist/Cache.d.ts:1:29 - error TS2307: Cannot find module 'apollo-cache' or its corresponding type declarations.

1 import { ApolloCache } from 'apollo-cache';
                              ~~~~~~~~~~~~~~

node_modules/apollo-cache-persist/types/index.d.ts:1:29 - error TS2307: Cannot find module 'apollo-cache' or its corresponding type declarations.

1 import { ApolloCache } from 'apollo-cache';
                              ~~~~~~~~~~~~~~
```

```
 "@apollo/client": "^3.1.3",
 "apollo-cache-persist": "^0.1.1",
```

**Comment by adsee42** [comment]
Hi, @robertvorthman 
I tried all example codes include yours, but nothing showed up in chrome dev tool localstorage.
And `console.log(localstorage)` showed nothing neither.

Any idea why is that?

**Comment by wtrocki** [comment]
Support landed in master. Working on releasing this
